boasiHQ/interactive-inputs
# [I-1] Allow users to more easily copy target value for select multiselect field types

## Context:

Link to context:  
  - [I-1](https://github.com/boasiHQ/interactive-inputs/issues/1).

## What has been done:

- Updated field properties to parse `disableAutoCopySelection`
- Added logic in the portal landing page to handle copying the passed value to the clipboard and pushing toast notification
- Updated `multiselect` and select divs to be Alpine JS enabled
- Added logic to make sure that the user can turn off auto copy on select
- Updated getting started doc to include reference to the new property on `select`/`multiselect` field
- Updated main README to have details on the `disableAutoCopySelection` field property

## Tests

- Clone this repo and check out this branch, `ii-i1-allow-users-to-more-easily-copy-target-value-for-select-multiselect-field-types`
- Make sure you have ASDF installed, the relevant plugins and target packages - `asdf install`
- Follow the steps in the [`getting-started.md` file in the `## Usage` section](https://github.com/boasiHQ/interactive-inputs/blob/ii-i1-allow-users-to-more-easily-copy-target-value-for-select-multiselect-field-types/gettting-started.md#usage)
  - Remember to remove the line containing `IAIP_SKIP_CONFIG_PARSE=1` or set it to `0`  before running
- Run the command from the step above
- Visit: http://localhost:8080/
- You should be able to see that your choice for "Favourite car" is copied to your  clipboard

## Screenshots

- Add field with new default behaviour active:

```yaml
...
          title: 'A batch of 10 feature flags have been added to be deployed. Would you like to proceed?'
          interactive: |
            fields:
             ...
              - label: exclude-domains
                properties:
                  description: Select the domains to exclude from the roll out
                  display: Exclude domain(s)
                  type: multiselect
                  choices: 
                    ["Payments", "Bookings", "Notifications", "Support"]
...
```

- ![InteractiveInput_CopyToClipboard](https://github.com/user-attachments/assets/779dc1b1-c623-43ec-a13e-145e50a52da1)

- <img width="830" alt="image" src="https://github.com/user-attachments/assets/18028267-a931-41b7-9eec-551d25c27503">
